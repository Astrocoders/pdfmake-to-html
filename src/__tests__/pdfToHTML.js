import pdfToHTML from '../index'

test('should properly convert a pdf doc definition to DOM', () => {
  const documentDefinition = [{
    stack: [
      {text: 'Text'},
      {text: 'Text'},
      {
        table: {
          body: [
            ['Column\nBreak', 'Column', 'Column'],
            ['Column', 'Column', 'Column'],
            ['Column', 'Column', {text: 'Column', bold: true}],
            ['Column', 'Column', {
              stack: [
                {text: 'Stacked cell'},
                {text: 'Stacked cell'},
                {text: 'Stacked cell'},
              ]
            }],
          ],
        },
      }
    ]
  }]

  expect(pdfToHTML({pdfDocument: documentDefinition}).innerHTML).toMatchSnapshot()
})
