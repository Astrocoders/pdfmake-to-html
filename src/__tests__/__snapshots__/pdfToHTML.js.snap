// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly convert a pdf doc definition to DOM 1`] = `"<div class=\\"ui sixteen column wide\\"><div class=\\"ui sixteen column wide\\"><p style=\\"\\">Text</p></div><div class=\\"ui sixteen column wide\\"><p style=\\"\\">Text</p></div><div class=\\"ui sixteen column wide\\"><div class=\\"ui three columns grid\\"><div class=\\"row\\"><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div></div><div class=\\"row\\"><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div></div><div class=\\"row\\"><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"font-weight: bold;\\">Column</p></div></div><div class=\\"row\\"><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><p style=\\"\\">Column</p></div><div class=\\"column\\"><div class=\\"ui sixteen column wide\\"><div class=\\"ui sixteen column wide\\"><p style=\\"\\">Stacked cell</p></div><div class=\\"ui sixteen column wide\\"><p style=\\"\\">Stacked cell</p></div><div class=\\"ui sixteen column wide\\"><p style=\\"\\">Stacked cell</p></div></div></div></div></div></div></div>"`;
